### PR TITLE
DDF-1543 UI changes that accompany fix to USNG

### PR DESCRIPTION
**Pull request by harrison-tarr**
Code changes to UI to accommodate the fixes to the USNG library that are a part of this ticket. 
https://github.com/codice/usng.js/pull/4
@AzGoalie @mverret @shaundmorris @brendan-hofmann 
@andrewkfiedler is hero

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/225)

### CODE REVIEW

**Comment by kcwire** [comment]
Please provide a description in the title of the PR

**Comment by shaundmorris** [comment]
Is there any kind of automated testing we can do on this to prevent potential regressions?


---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/225)

**Comment by harrison-tarr** [comment]
travis is working on tests for this code - this is intended to be a 'living PR' giving a chance for reviewers to begin looking at the code


---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/225)

**Comment by shaundmorris** [comment]
Sweet thanks @harrison-tarr


---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/225)

**Comment by harrison-tarr** [comment]
@andrewkfiedler to hero this PR please:
1. do a normal PRBS
2. copy this file: https://github.com/AzGoalie/usng.js/blob/Issue-3/usng.js
3. replace your `usng.js` in `/ddf/catalog/ui/search-ui/standard/target/lib` with this file
4. run `mvn test` in `/ddf/catalog/ui/search-ui/standard`

if all of the new tests pass, it's good to go (there will likely be old, unrelated tests that do not pass)

@AzGoalie feel free to correct me if I haven't accurately described how to test

**Comment by brendan-hofmann** [comment]
Reviewed 1 of 3 files at r1, 3 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js, line 97 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/225#-K-stjzYwQwhJZnRM7kt)** ([raw file](https://github.com/codice/ddf/blob/7d5bf4196c4f85fecdcd588881f411f395b2dd76/catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js#L97)):</sup>
You set these values inside a conditional here. Do you need to guard against them being null elsewhere?


---

<sup>**[catalog/ui/search-ui/standard/src/test/js/wd/usng.js, line 38 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/225#-K-su79jOhijpr0q-Sww)** ([raw file](https://github.com/codice/ddf/blob/7d5bf4196c4f85fecdcd588881f411f395b2dd76/catalog/ui/search-ui/standard/src/test/js/wd/usng.js#L38)):</sup>
Can you elaborate on what kind of edge cases these tests are exercising in some comments? Otherwise they just look like random numbers.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/225)

**Comment by harrison-tarr** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js, line 97 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/225#-K-stjzYwQwhJZnRM7kt)** ([raw file](https://github.com/codice/ddf/blob/7d5bf4196c4f85fecdcd588881f411f395b2dd76/catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js#L97)):</sup>
no, it's only used in that `set`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/225)

**Comment by andrewkfiedler** [comment]
:+1:  built and tested

File: pom.xml [hunk 1]
```diff
@@ -58,7 +58,7 @@
         <reports.repository.url />
 
         <!-- Must MANUALLY update this if the ddf/support project's version changes -->
-        <ddf.support.version>2.3.4-SNAPSHOT</ddf.support.version>
+        <ddf.support.version>2.3.4.ALPHA2</ddf.support.version>
```
Comment (shaundmorris) [inline pom.xml]: Interesting why are we depending on an ALPHA version?

**Comment by ahoffer** [comment]
Built both DDF and DIB successfully.
Harrison walk me through putting the new javascript into the usng.js file.
The new code passed all the USNG unit tests executed by grunt.

**Pull request merged**